### PR TITLE
support lbmethod modules

### PR DESCRIPTION
**Pull request by xorpaul**
`mod_proxy_balancer` depends on one of these load balancing scheduler algorithms:
* [mod_lbmethod_byrequests](https://httpd.apache.org/docs/2.4/mod/mod_lbmethod_byrequests.html)
* [mod_lbmethod_bytraffic](https://httpd.apache.org/docs/2.4/mod/mod_lbmethod_bytraffic.html)
* [mod_lbmethod_bybusyness](https://httpd.apache.org/docs/2.4/mod/mod_lbmethod_bybusyness.html)
* [mod_lbmethod_heartbeat](https://httpd.apache.org/docs/2.4/mod/mod_lbmethod_heartbeat.html)

See https://httpd.apache.org/docs/2.4/mod/mod_proxy_balancer.html

This adds support to include one of them via:
```
class { 'apache::mod::lbmethod_byrequests': }
```

Maybe `apache::mod::lbmethod_byrequests` should even fail if none of these 4 are included?

### CODE REVIEW

File: manifests/mod/lbmethod_bybusyness.pp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# 
+# @see https://httpd.apache.org/docs/2.4/mod/mod_lbmethod_bybusyness.html for additional documentation.
+#
+class apache::mod::lbmethod_bybusyness (
+  Optional[String] $apache_version   = $apache::apache_version,
+) {
+  require apache::mod::proxy_balancer
+
+  if versioncmp($apache_version, '2.3') >= 0 {
+    ::apache::mod { 'lbmethod_bybusyness': }
```
Comment (ekohl) [inline manifests/mod/lbmethod_bybusyness.pp:15]: This is true for all files, but it's no longer needed to prefix with `::`
```suggestion
    apache::mod { 'lbmethod_bybusyness': }
```

**Review by ekohl — COMMENTED** [review]
It would be good to have some basic unit tests for these classes.

File: manifests/mod/lbmethod_bybusyness.pp (hunk 1 above)
Comment (xorpaul) [reply manifests/mod/lbmethod_bybusyness.pp]: removed unnecessary prefix

**Review by ekohl — COMMENTED** [review]
Minimal specs in https://github.com/puppetlabs/puppetlabs-apache/tree/main/spec/classes/mod would be nice to have. That way at least the catalog is compiled and verified.

**Comment by xorpaul** [comment]
@ekohl Ok, thanks. I'll check it out.

**Comment by xorpaul** [comment]
Is there any way to trigger only my new tests and not the whole test suite which takes over 6 minutes on my workstation?

**Comment by david22swan** [comment]
@xorpaul You can target specific test files as shown below:
```
bundle exec rspec spec/unit/puppet/provider/iptables_spec.rb
```

**Comment by david22swan** [comment]
@xorpaul Hey, just checking in on how this is progressing

**Comment by xorpaul** [comment]
After upgrading my workstation to Ubuntu 22.04 the bundle commands do not work anymore.

I'm getting
```
$ bundle install
Could not find gem 'puppet-module-posix-default-r3.0 (~> 1.0)' in any of the gem sources listed in your Gemfile.
$ bundle exec rake parallel_spec
/usr/bin/env: ‘ruby2.5’: No such file or directory
```

Is there a way to make this work with more recent OS versions?

**Comment by david22swan** [comment]
@xorpaul You could try deleting the gem folder and gemfile.lock file

**Comment by ekohl** [comment]
@xorpaul I recognize the pain you have. Sadly PDK is in a horrible place regarding modern Ruby support. https://tickets.puppetlabs.com/browse/MODULES-11161 has been open for a long time. I always end up hacking `Gemfile` and hardcode `minor_version = '2.7'` here:
https://github.com/puppetlabs/puppetlabs-apache/blob/648332c61deb3d240a0d14af034d8e59f9f6197e/Gemfile#L17

Or I just rely on CI to do testing.

**Comment by david22swan** [comment]
@xorpaul The issue you were having should now be resolved.

**Review by david22swan — APPROVED** [review]
LGTM
@xorpaul Sorry for the wait, this just sort of slipped out of mind.
Thank's for getting the work done.

**Pull request merged**